### DISCUSSION TITLE
How do you toggle row counts on tables?

### DISCUSSION THREAD

**Discussion by ninmonkey** [dfinke/ImportExcel#1295] [Q&A] [answered]
I'm creating a worksheet `foo` with a table `bar` . How do I

- click a table => toggle `total row` on

![image](https://user-images.githubusercontent.com/3892031/198389840-a6fb4822-f231-45b4-b3e2-0a06a2eb257b.png)

- and set at least one column to `count` ?

![image](https://user-images.githubusercontent.com/3892031/198389895-05f01c2a-3b15-4dba-95c6-ec54becfd941.png)

**Comment by stahler** [comment]
Not sure if this is what you are looking for, but it may be a start for you.
```powershell
$data = ConvertFrom-Csv @"
Region,State,Units,Price
West,Texas,927,923.71
North,Tennessee,466,770.67
East,Florida,520,458.68
East,Maine,828,661.24
"@

$path = 'c:\temp\test1.xlsx'
Remove-Item -Path $path -ErrorAction SilentlyContinue

$pkg = $data | Export-Excel -Path $path -PassThru

Add-ExcelTable -Range $pkg.Sheet1.Cells["A1:D5"] -TableStyle Medium6 -TotalSettings @{Region="Count"} 
Close-ExcelPackage $pkg -Show

**Comment by ninmonkey** [comment, marked as answer]
[That got me there. here's a version without a commandlet](https://github.com/dfinke/ImportExcel/discussions/1328)
```ps1
[OfficeOpenXml.ExcelPackage]$pkg2 = Open-ExcelPackage $Path.Dest
[OfficeOpenXml.Table.ExcelTable]$table1 = $pkg2.Sheet1.Tables[0]
$rowFunc = [OfficeOpenXml.Table.RowFunctions]

$table1.Columns[0].TotalsRowFunction = $rowFunc::Count
$table1.Columns[2].TotalsRowFunction = $rowFunc::Average
```

**Reply by dfinke** [reply]
Why all the ceremony and casting the variables?
